mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -271,18 +277,25 @@ public RxBleDevice getBleDevice(@NonNull String macAddress) {
         return createScanOperation(filterServiceUUIDs);
     }
 
+    public void simulateDeviceDiscovery(@NonNull RxBleDeviceMock deviceMock) {
```

Comment:
One more thing. The method name suggests that this is one time operation, I mean that it will emit provided device and will not affect future scans. By use of a ReplaySubject each device is stored and then magically replied during following scan operations.

Maybe we could have to separate configurations. Static - configured when the mock is instantiated (stored in discoverableDevices) and dynamic, on demand. You may use a PublishSubject instead and do following in line #291

`
discoveredDevicesSubject
.startWith(Observable.from(discoverableDevices))
`